### PR TITLE
fix(50415): Language server debug failure - Did not expect GetAccessor to have an Identifier in its trivia

### LINKED ISSUE #50415: Language server debug failure - Did not expect GetAccessor to have an Identifier in its trivia
# Bug Report

### 🔎 Search Terms

- auto complete
- suggestions
- snippets
- language server

### 🕗 Version & Regression Information

- This changed between versions 4.6.3 and 4.6.4

### ⏯ Playground Link

N/A, requires npm dependency

### 💻 Code

Code sample unfortunately uses an external library as I've not been able to nail down an exact repro, but this issue only seems to occur in >=4.6.4 so I think this is less likely to be a purely type definition issue. Also reproduces in the latest dev version (at posting, 4.9.0-dev.20220823). I've also seen this error with SetAccessor instead of GetAccessor in similar situations.

#### Repro steps

1. New project/workspace in vscode
2. `npm init -y`.
3. `npm i @pixi/display@6.5.1`
4. `npm i -D typescript@4.6.3`
5. Create the following file as `src/index.ts`:
```ts
import { DisplayObject } from "@pixi/display";

class A extends DisplayObject
{
	
}
```
6. Place cursor within the class definition
7. CTRL+Space to get suggestions/auto complete/snippets
8. Change TS version to 4.6.4, repeat steps 6 & 7.
9. No suggestions/auto complete/snippets this time.

### 🙁 Actual behavior

No suggestions/snippets provided, extension host log contains the following:
```
[2022-08-23 11:55:41.066] [exthost] [error] Error: <semantic> TypeScript Server Error (4.9.0-dev.20220823)
Debug Failure. Did not expect GetAccessor to have an Identifier in its trivia
Error: Debug Failure. Did not expect GetAccessor to have an Identifier in its trivia
    at addSyntheticNodes (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:166443:30)
    at processNode (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:166416:13)
    at visitNode (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:30934:24)
    at forEachChildInGetAccessor (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:31105:17)
    at Object.forEachChild (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:31714:47)
    at NodeObject.forEachChild (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:166397:23)
    at createChildren (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:166431:14)
    at NodeObject.getChildren (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:166374:56)
    at find (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:129452:30)
    at Object.findPrecedingToken (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:129445:22)
    at formatSpanWorker (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:149386:44)
    at d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:149309:140
    at Object.getFormattingScanner (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:147936:23)
    at Object.formatNodeGivenIndentation (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:149309:31)
    at d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134870:42
    at Object.flatMap (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:524:25)
    at Object.printAndFormatSnippetList (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134868:34)
    at getEntryForMemberCompletion (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134672:42)
    at createCompletionEntry (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134492:23)
    at getCompletionEntriesFromSymbols (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134997:29)
    at completionInfoFromData (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134304:31)
    at Object.getCompletionsAtPosition (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:134164:36)
    at Object.getCompletionsAtPosition (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:167677:35)
    at IpcIOSession.Session.getCompletions (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:179809:64)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:178482:61)
    at d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:180542:96
    at IpcIOSession.Session.executeWithRequestId (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:180533:28)
    at IpcIOSession.Session.executeCommand (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:180542:41)
    at IpcIOSession.Session.onMessage (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:180570:35)
    at process.<anonymous> (d:\Temp\snippetbugtest\node_modules\typescript\lib\tsserver.js:184736:31)
    at process.emit (node:events:390:28)
    at emit (node:internal/child_process:917:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
	at Function.create (c:\Users\Russell.Speke\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:509517)
	at dispatchResponse (c:\Users\Russell.Speke\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:503365)
	at dispatchMessage (c:\Users\Russell.Speke\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:502233)
	at ChildProcess.<anonymous> (c:\Users\Russell.Speke\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:501718)
	at ChildProcess.emit (node:events:390:28)
	at emit (node:internal/child_process:917:12)
	at processTicksAndRejections (node:internal/process/task_queues:84:21)
```

### 🙂 Expected behavior

Auto complete/snippets etc should work as normal.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #50415

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Can you amend the test case to test both orders of accessors as in

https://github.com/microsoft/TypeScript/commit/20432a64e32d6311a8302362d1185dae9d296e08

and

and also clone the nodes for the set-accessor even if it's technically unnecessary?

https://github.com/microsoft/TypeScript/commit/b1358db5d1b6cf73ca97a1d7e3577e45e5eae303

**Comment by a-tarasyuk** [comment]
> and also clone the nodes for the set-accessor even if it's technically unnecessary?

@DanielRosenwasser  We need to clone nodes only for one of them - `get` or `set`. We have one copy of the properties applied to two methods simultaneously, which causes the error. Most of the props are new copies, so we just need to create new _copies_ for one of them
https://github.com/microsoft/TypeScript/blob/07157914eb26a6e9abdcda7255c4094c9efff4cd/src/services/codefixes/helpers.ts#L82-L84

**Comment by DanielRosenwasser** [comment]
I understand that - I guess it doesn't entirely matter, but there are so many defensive copies all over the method other than in the updated section of code that I felt like it was probably better to be consistent than risk someone else glancing at the one place where we don't do a defensive copy.

If you're open to it, I was thinking of switching `name`, `modifiers`, and `body` to be local helper functions that generate unique deep clones for each usage. I'm not whether others would see it as an improvement.

**Review by gabritto — APPROVED** [review]

**Comment by gabritto** [comment]
Looks good to me, thanks. I don't really have an opinion on what Daniel suggested, so it's up to you and him.

**Comment by a-tarasyuk** [comment]
@gabritto I'll add updates

File: src/compiler/factory/nodeFactory.ts [hunk 1]
```diff
@@ -6839,6 +6840,7 @@ namespace ts {
             }
         }
         if (startsOnNewLine !== undefined) destEmitNode.startsOnNewLine = startsOnNewLine;
+        if (snippetElement !== undefined) destEmitNode.snippetElement = snippetElement;
```
Comment (DanielRosenwasser) [inline src/compiler/factory/nodeFactory.ts:6843]: What exactly does the `snippetElement` property do? I didn't really see need this change on my branch.

Comment (DanielRosenwasser) [reply src/compiler/factory/nodeFactory.ts:6843]: Oh I guess this is for placing the cursor in the right place - but is there an easy way to show this not working right now?

Comment (a-tarasyuk) [reply src/compiler/factory/nodeFactory.ts:6843]: It needs to clone this snippet. We need this when we use _snippets_
https://github.com/microsoft/TypeScript/blob/71d19115031ae1c48fa34a24636dfae5e87e1b08/src/services/completions.ts#L937

Comment (a-tarasyuk) [reply src/compiler/factory/nodeFactory.ts:6843]: In the current implementation we don’t use clone for `body`

Comment (a-tarasyuk) [reply src/compiler/factory/nodeFactory.ts:6843]: If we try to clone the body without these changes, we lose the snippet - for example, tests with enabled `includeCompletionsWithSnippetText` will fail (`completionsOverridingMethod2`, etc.)

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**